### PR TITLE
[MM-16168] Migrate Team.GetAllTeamListing to Sync by default

### PR DESCRIPTION
**Pull request by mounicapaladugu**
#### Summary
Migrate "Team.GetAllTeamListing" to Sync by default

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16168

Fixes https://github.com/mattermost/mattermost-server/issues/11086

### CODE REVIEW

**Comment by mounicapaladugu** [comment]
I just reverted GetAllTeamPageListing error message as i18n check was failing, but the error registered with that function doesn't seem to match the function definition.

File: app/team.go [hunk 1]
```diff
@@ -641,11 +641,11 @@ func (a *App) GetAllPrivateTeamsPage(offset int, limit int) ([]*model.Team, *mod
 }
 
 func (a *App) GetAllPublicTeams() ([]*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().GetAllTeamListing()
```
Comment (jespino) [inline app/team.go]: We can simplify here using just `return a.Srv.Store.Team().GetAllTeamListing()`

**Comment by jespino** [comment]
@mounicapaladugu I liked the change proposed by you for the error string. So, you deserves an explanation that I didn't give to you properly before. You can change this kind of strings, and add new ones, and things like that, when the i18n error happens in jenkins, what it means it that you have introduced/changed/removed any translation string. So... when you do that, you have to execute `make i18n-extract` to extract the strings from the source code to the file `i18n/en.json` (after executing the `make i18n-extract` command you will see the changes in that file). Depending on the changes, you probably have to review manually the included changes, for example, if you added a new error type, probably you have to add there the real text expected to be shown when this error happens. So... feel free to re-add the change and do the other changes if you want :)

**Comment by jespino** [comment]
@mounicapaladugu This is almost done, do you need anything from our side?

**Comment by jespino** [comment]
I'm taking over this PR to finish it and merge it.

**Comment by cpoile** [comment]
@jespino merging blocked on you. Just fyi. :)

**Pull request merged**